### DISCUSSION TITLE
Is event logging to a file supported?

### DISCUSSION THREAD

**Discussion by ramneekers** [keycloak/keycloak#11529] [Archive]
### Describe the bug

Hello everyone,

We are running a Kubernetes cluster that uses the Wildfly version of Keycloak, and we write our `Login` and `Admin` events to a sidecar for security. We do this via a startup script defined in `values.yaml`.

```
      embed-server --server-config=standalone-ha.xml --std-out=echo
      batch
      /subsystem=logging/json-formatter=json-formatter:add(exception-output-type=detailed-and-formatted, print-details=false)
      /subsystem=logging/file-handler=EVENTLOG:add(file={"path"=>"/var/log/events.log"})
      /subsystem=logging/file-handler=EVENTLOG:write-attribute(name="named-formatter", value="json-formatter")
      /subsystem=logging/logger=org.keycloak.events:add(use-parent-handlers="false", level=INFO, handlers=[EVENTLOG])
      /subsystem=keycloak-server/spi=eventsListener:add
      /subsystem=keycloak-server/spi=eventsListener/provider=jboss-logging:add(enabled=true)
      /subsystem=keycloak-server/spi=eventsListener/provider=jboss-logging:write-attribute(name=properties.success-level,value=info)
      /subsystem=keycloak-server/spi=eventsListener/provider=jboss-logging:write-attribute(name=properties.error-level,value=warn)
      run-batch
      stop-embedded-server
```

As of version 17.0.1, we are migrating to the Quarkus distribution. Since startupScripts are no longer used due to the CLI being deprecated, how do we replicate this behavior?

### Version

17.0.1

### Expected behavior

Keycloak writes `Login` and `Admin` events to the `events.log` file in the mounted volume.

### Actual behavior

Keycloak cannot find the file, and logs nothing.

**Comment by chrobotm** [comment]
bump

**Comment by psytester** [comment]
I don't know a build-in solution, but I have setup a native qarkus logging.
Setup in `conf/quarkus.properties` all logging you need.
Console, file and an additional handler just with the events channel.
It writes all events to a logfile.

vi conf/keycloak.conf
```quarkus-properties-enabled=true```

vi conf/quarkus.properties
```
# Logging root
quarkus.log.level=...

# Console logging
quarkus.log.console.level=INFO

# Configure a named handler that logs to Events file
quarkus.log.handler.file."EVENTS".enable=true
quarkus.log.handler.file."EVENTS".level=TRACE
#quarkus.log.handler.file."EVENTS".min-level=TRACE
quarkus.log.handler.file."EVENTS".path=/opt/logfiles/events.log
quarkus.log.handler.file."EVENTS".format=%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%c{3.}] (%t) %s%e%n
quarkus.log.category."org.keycloak.events".level=TRACE
quarkus.log.category."org.keycloak.events".handlers=EVENTS

## General log goes to Application file
quarkus.log.file.enable=true
quarkus.log.file.path=/opt/logfiles/keycloak.log
quarkus.log.file.format=%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%c{3.}] (%t) %s%e%n
quarkus.log.file.level=...
quarkus.log.min-level=...
```

**Comment by tijojosetom** [comment]
You can use this option:  `bin/kc.[sh|bat] start --log-level=INFO,org.keycloak.events:debug`
Details here [https://www.keycloak.org/server/logging]

**Comment by TorinKS** [comment]
Hi, is that correct approach to enable debug severity in PROD env if we would like to have login related events on pod stdout? It seemed perfect before, when we just propagated success events to INFO instead of DEBUG using JBoss cli, like
  /subsystem=keycloak-server/spi=eventsListener/provider=jboss-logging:write-attribute(name=properties.success-level,value=info)
Do we have a similar approach for Quarkus? I am wondering because I think theoretically enabling debug for org.keycloak.events may cause something more specific to dev env and may slow down performance

**Reply by dasniko** [reply]
Use config `spi-events-listener-jboss-logging-success-level=info` or env var `KC_SPI_EVENTS_LISTENER_JBOSS_LOGGING_SUCCESS_LEVEL=info`
For common info on the naming convention used here, see guide https://www.keycloak.org/server/configuration-provider#_configuration_option_format

**Reply by TorinKS** [reply]
Perfect! thanks!
